client/webserver/http.go

```diff
@@ -190,6 +195,126 @@ func (s *WebServer) handleOrders(w http.ResponseWriter, r *http.Request) {
 	})
 }
 
+// handleExportOrders is the handler for the /orders/export page request.
+func (s *WebServer) handleExportOrders(w http.ResponseWriter, r *http.Request) {
+	wf, ok := w.(http.Flusher)
+	if !ok {
+		log.Errorf("unable to flush streamed data")
+		http.Error(w, "unable to flush streamed data", http.StatusBadRequest)
+		return
+	}
+
+	filter := new(core.OrderFilter)
+	err := r.ParseForm()
+	if err != nil {
+		log.Errorf("error parsing form for export order: %v", err)
+		http.Error(w, http.StatusText(http.StatusBadRequest), http.StatusBadRequest)
+		return
+	}
+
+	filter.Hosts = r.Form["hosts"]
+	assets := r.Form["assets"]
+	filter.Assets = make([]uint32, len(assets))
+	for k, assetStrID := range assets {
+		assetNumID, err := strconv.ParseUint(assetStrID, 10, 32)
+		if err != nil {
+			log.Errorf("error parsing asset id: %v", err)
+			http.Error(w, http.StatusText(http.StatusBadRequest), http.StatusBadRequest)
+			return
+		}
+		filter.Assets[k] = uint32(assetNumID)
+	}
+	statuses := r.Form["statuses"]
+	filter.Statuses = make([]order.OrderStatus, len(statuses))
+	for k, statusStrID := range statuses {
+		statusNumID, err := strconv.ParseUint(statusStrID, 10, 16)
+		if err != nil {
+			http.Error(w, http.StatusText(http.StatusBadRequest), http.StatusBadRequest)
+			log.Errorf("error parsing status id: %v", err)
+			return
+		}
+		filter.Statuses[k] = order.OrderStatus(statusNumID)
+	}
+
+	ords, err := s.core.Orders(filter)
+	if err != nil {
+		log.Errorf("error retrieving order: %v", err)
+		http.Error(w, http.StatusText(http.StatusInternalServerError), http.StatusInternalServerError)
+		return
+	}
+
+	w.Header().Set("Content-Disposition", "attachment; filename=orders.csv")
+	w.Header().Set("Content-Type", "text/csv")
+	w.WriteHeader(http.StatusOK)
+	csvWriter := csv.NewWriter(w)
+	csvWriter.UseCRLF = strings.Contains(r.UserAgent(), "Windows")
+
+	err = csvWriter.Write([]string{
+		"Host",
+		"Base",
+		"Quote",
+		"Ask",
+		"Offer",
+		"Type",
+		"Side",
+		"Time in Force",
+		"Status",
+		"Rate",
+		"Filled (%)",
+		"Settled (%)",
+		"Quantity",
+		"Time",
+	})
+	if err != nil {
+		log.Errorf("error writing CSV: %v", err)
+		return
+	}
+	csvWriter.Flush()
+	err = csvWriter.Error()
+	if err != nil {
+		log.Errorf("error writing CSV: %v", err)
+		return
+	}
+	wf.Flush()
+
+	for _, ord := range ords {
+		ordReader := orderReader{ord}
+		timestamp, err := encode.UnixTimeMilliLocal(int64(ord.Stamp)).MarshalText()
```

Comment:
`MarshalText` just applies `RFC3339Nano` formatting, so i'd suggest just doing `.Format(time.RFC3339Nano)`.  RFC3339 is not the most human friendly format, but it's standardized and good for parsing, so I think for CSV it's good.

Also, since we know these stamps are only millisecond precision, we could specify the format accordingly like `2006-01-02T15:04:05.999Z07:00`.